src/Base/Core/Quality/IQuality.hs

```diff
@@ -48,6 +50,21 @@ instance Show Quality where
   show (Diminished i) = if i == 1 then "dim" else show i ++ "dim"
   show (Augmented i)  = if i == 1 then "aug" else show i ++ "aug"
 
+major :: Quality
+major = Major
+
+perfect :: Quality
+perfect = Perfect
+
+minor :: Quality
+minor = Minor
+
+diminished :: Int -> Maybe Quality
+diminished x = if x > 0 && x < 12 then Just $ Diminished x else Nothing
+
+augmented :: Int -> Maybe Quality
+augmented x = if x > 0 && x < 12 then Just $ Augmented x else Nothing
```

Comment:
So when I say "smart constructors", for a sum type like `Quality`, this is what I mean: One function per constructor in the `data` declaration that enforces invariants of that constructor. Your original smart constructor, `iQualFrom`, would have been appropriate if this was a _product_ type, such as any of the various record types we've introduced so far; since it's a type with variants, you need at least one function per variant.